src/adapters/curve/products/reward/curveRewardAdapter.ts

```diff
@@ -0,0 +1,130 @@
+import { IMetadataBuilder } from '../../../../core/decorators/cacheToFile'
+import { filterMapAsync } from '../../../../core/utils/filters'
+import {
+  ProtocolDetails,
+  PositionType,
+  GetPositionsInput,
+  ProtocolPosition,
+  TokenType,
+  GetEventsInput,
+  MovementsByBlock,
+} from '../../../../types/adapter'
+import { CrvMinter__factory, StakingContract__factory } from '../../contracts'
+import { CurveStakingAdapter } from '../staking/curveStakingAdapter'
+
+const CRV_MINTER_ADDRESS = '0xd061D61a4d941c39E5453435B6345Dc261C2fcE0'
+
+const CRV_TOKEN = {
+  address: '0xd533a949740bb3306d119cc777fa900ba034cd52',
+  name: 'Curve DAO Token',
+  symbol: 'CRV',
+  decimals: 18,
+  iconUrl:
+    'https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/crv.svg',
+}
+
+export class CurveRewardAdapter
+  extends CurveStakingAdapter
+  implements IMetadataBuilder {
+  productId = 'reward'
+
+  /**
+   * Update me.
+   * Add your protocol details
+   */
+  getProtocolDetails(): ProtocolDetails {
+    return {
+      protocolId: this.protocolId,
+      name: 'Curve reward',
+      description: 'Curve reward adapter',
+      siteUrl: 'https://curve.fi/',
+      iconUrl:
+        'https://raw.githubusercontent.com/MetaMask/contract-metadata/master/images/crv.svg',
+      positionType: PositionType.Reward,
+      chainId: this.chainId,
+      productId: this.productId,
+    }
+  }
+
+  async getPositions({
+    userAddress,
+    blockNumber,
+  }: GetPositionsInput): Promise<ProtocolPosition[]> {
+    const stakingContracts = await this.getProtocolTokens()
+
+    return await filterMapAsync(stakingContracts, async (protcolToken) => {
+
+      const { address } = protcolToken
+
+
+      const stakingContract = StakingContract__factory.connect(
+        address,
+        this.provider,
+      )
+
+      const balanceRaw = await stakingContract.claimable_tokens.staticCall(
+        userAddress,
+        { blockTag: blockNumber },
+      ).catch(e => 0n)
+
```

Comment:
Latest curve pools (which have low liquidity) have a different reward interface. So this requests fails for those pools. 

Since with this MR we have feature parity with our old adapters I think we should manage new curve-rewards under a different adapter.

Reply:
Upon further inspection, they do have this method, it fails if you run the tests at a block number before the contract was deployed

Reply:
catching and returning 0 I think is not a bad solution...